// [[Reason: Expando function declarations are not fixed.]] ////

//// [tests/cases/compiler/isolatedDeclarationErrorsExpandoFunctions.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -2,11 +2,45 @@
 
 //// [isolatedDeclarationErrorsExpandoFunctions.d.ts]
 export declare function foo(): void;
 export declare namespace foo {
-    var apply: () => void;
-    var call: () => void;
-    var bind: () => void;
-    var caller: () => void;
-    var toString: () => void;
-    var length: number;
+    var apply: invalid;
+    var call: invalid;
+    var bind: invalid;
+    var caller: invalid;
+    var toString: invalid;
+    var length: invalid;
 }
+
+/// [Errors] ////
+
+isolatedDeclarationErrorsExpandoFunctions.ts(3,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+isolatedDeclarationErrorsExpandoFunctions.ts(4,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+isolatedDeclarationErrorsExpandoFunctions.ts(5,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+isolatedDeclarationErrorsExpandoFunctions.ts(6,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+isolatedDeclarationErrorsExpandoFunctions.ts(7,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+isolatedDeclarationErrorsExpandoFunctions.ts(8,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+
+
+==== isolatedDeclarationErrorsExpandoFunctions.ts (6 errors) ====
+    export function foo(): void {}
+    
+    foo.apply = () => {}
+    ~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.call = ()=> {}
+    ~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.bind = ()=> {}
+    ~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.caller = ()=> {}
+    ~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.toString = ()=> {}
+    ~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.length = 10
+    ~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+    foo.length = 10
+    
\ No newline at end of file
